hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/StoreFileWriter.java

```diff
@@ -242,6 +248,29 @@ public void appendMetadata(final long maxSequenceId, final boolean majorCompacti
     appendTrackedTimestampsToMetadata();
   }
 
+  /**
+   * Appends MOB - specific metadata (even if it is empty)
+   * @param mobRefSet - set of MOB file names
+   * @throws IOException problem writing to FS
+   */
+  public void appendMobMetadata(Set<String> mobRefSet) throws IOException {
+    if (mobRefSet != null && mobRefSet.size() > 0) {
+      StringBuilder sb = new StringBuilder(2 * mobRefSet.size() - 1);
+      String[] arr = new String[mobRefSet.size()];
```

Comment:
this seems overly complicated. either use common-lang's `StringUtils.join` or just iterate over the `Set`.